src/test/shared/featureController.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { FeatureController } from '../../shared/featureController'
+import { TestSettingsConfiguration } from '../utilities/testSettingsConfiguration'
+
+describe('FeatureController', async () => {
+
+    const testConfig = new TestSettingsConfiguration()
+    const activeSettingKey = 'activeSetting'
+    const inactiveSettingKey = 'inactiveSetting'
+    const missingSettingKey = 'missingSetting'
+    const variableSettingKey = 'variableSetting'
+    const permanentSettingKey = 'permanentSetting'
+    const permanentMissingSettingKey = 'permanentMissingSetting'
+    const permanentMissingVariableSettingKey = 'permanentMissingVariableSetting'
+    await testConfig.writeSetting(`toggle.${activeSettingKey}`, true)
+    await testConfig.writeSetting(`toggle.${inactiveSettingKey}`, false)
+    await testConfig.writeSetting(`toggle.${variableSettingKey}`, false)
+    await testConfig.writeSetting(`toggle.${permanentSettingKey}`, true)
+
+    const features = new FeatureController(testConfig, [
+        permanentSettingKey,
+        permanentMissingSettingKey,
+        permanentMissingVariableSettingKey
+    ])
+
+    it('returns whether or not a feature is active', () => {
+        assert.strictEqual(features.isFeatureActive(activeSettingKey), true)
+        assert.strictEqual(features.isFeatureActive(inactiveSettingKey), false)
+    })
+
+    it('returns false for features that are not present', () => {
+        assert.strictEqual(features.isFeatureActive(missingSettingKey), false)
+    })
+
+    it('returns the current value of non-session-permanent features', async () => {
+        assert.strictEqual(features.isFeatureActive(variableSettingKey), false)
+        await testConfig.writeSetting(`toggle.${variableSettingKey}`, true)
```

Comment:
Because you mutate "toggle.${variableSettingKey}",  I would initialize it here. Consider removing the initialization above too, since it isn't used in any other test.

Reply:
Moved the initialization to the beforeEach call. I want to ensure that this is in the FeatureController when initializing that (otherwise this test essentially turns into the next test, where you're testing with a non-present feature).